src/components/List.tsx

```diff
@@ -7,9 +7,9 @@ const BlogList: React.FC<BlogListProps> = ({ blogPosts }) => {
   const [currentPage, setCurrentPage] = useState<number>(1);
   const [postsPerPage] = useState<number>(10);
 
-  const indexOfLastPost = currentPage * postsPerPage;
-  const indexOfFirstPost = indexOfLastPost - postsPerPage;
-  const currentPosts = blogPosts.slice(indexOfFirstPost, indexOfLastPost);
+  const indexOfLastPost = React.useMemo(() => currentPage * postsPerPage, [currentPage, postsPerPage]);
+  const indexOfFirstPost = React.useMemo(() => indexOfLastPost - postsPerPage, [indexOfLastPost, postsPerPage]);
+  const currentPosts = React.useMemo(() => blogPosts.slice(indexOfFirstPost, indexOfLastPost), [indexOfFirstPost, indexOfLastPost]);
```

Comment:
When you have values that are recomputed on every re-render it's recommended you memoize them. In this case it might not be a computationally difficult calculation to repeat on every re-render but it's something to keep an eye out for